src/main/java/com/jfrog/ide/common/nodes/SortableChildrenTreeNode.java

```diff
@@ -0,0 +1,17 @@
+package com.jfrog.ide.common.nodes;
+
+import org.apache.commons.collections.CollectionUtils;
+
+import javax.swing.tree.DefaultMutableTreeNode;
+import java.util.Comparator;
+
+public class SortableChildrenTreeNode extends DefaultMutableTreeNode {
```

Comment:
You can move the Comparator implementation declaration to here:
```suggestion
public abstract class SortableChildrenTreeNode<T> extends DefaultMutableTreeNode implements Comparator<T> {
```

By that you would benefit from the following:
1. Remove code duplication - you write `implements Comparator` only once.
2. Enforce implementation of `compareTo` on all child objects.